Include/internal/pycore_pystate.h

```diff
@@ -226,6 +226,7 @@ typedef struct pyruntimestate {
            If that becomes a problem later then we can adjust, e.g. by
            using a Python int. */
         int64_t next_id;
+        int finalizing;
```

Comment:
As yet another potential colour for this bikeshed:

```suggestion
        int allow_new;
```

That's pretty self-explanatory as to what it means (when it's false, then Py_NewInterpreter will fail), while still being short and to the point.

Reply:
I like this, and have gone with this at least for now :)